### DISCUSSION TITLE
Plans to increase client support for loading/including counters?

### DISCUSSION THREAD

**Discussion by georgiosd** [ravendb/ravendb#14601] [General]
Hey folks!

Any plans to increase support for loading/including counters?

Specifically:
- Include with `Load(string[])`
- Include with lazy `Load*`
- Load counter(s) lazily

**Comment by garayx** [comment]
Hi

1. I think you can already do this
`session.Include<User>(x => x.PartnerId).Load<User>(new[] { "users/1-A", "users/2-A" });`

2. What do you mean by `Load*`? There is a include for lazy load:
`session.Advanced.Lazily
                        .Include<User>(x => x.PartnerId)
                        .Load<User>("users/2-A");`

3. I opened a issue to implement that:
https://issues.hibernatingrhinos.com/issue/RavenDB-18999/Add-option-to-get-counter-and-timeseries-lazily

thanks for suggestion

**Comment by georgiosd** [comment]
Hey @garayx this is about *counters*, not documents (for all 3 items above). 😄

**Comment by garayx** [comment]
Sorry you are right, 

The 1st one you can do like this:
```
                    var user1 = newSession.Load<User>(new[] { "users/1", "users/2" }, i => i.IncludeAllCounters());
                    var user2 = newSession.Load<User>(new[] { "users/1", "users/2" }, i => i.IncludeCounter("clicks"));
                    var user3 = newSession.Load<User>(new[] { "users/1", "users/2" }, i => i.IncludeCounters(new string[] { "clicks", "references"}));
```

regarding the 2nd, we might add it in the future, https://issues.hibernatingrhinos.com/issue/RavenDB-19004/Add-option-to-include-counter-and-timeseries-in-lazy-load

**Reply by georgiosd** [reply]
Thank you @garayx !
